plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -963,17 +963,29 @@ func (c *PodAffinityChecker) InterPodAffinityMatches(pod *v1.Pod, meta algorithm
 	if node == nil {
 		return false, nil, fmt.Errorf("node not found")
 	}
-	if !c.satisfiesExistingPodsAntiAffinity(pod, meta, nodeInfo) {
-		return false, []algorithm.PredicateFailureReason{ErrPodAffinityNotMatch}, nil
+	if satisfy, error := c.satisfiesExistingPodsAntiAffinity(pod, meta, nodeInfo); !satisfy {
+		switch error {
```

Comment:
If you do what I said regarding returning algorithm.PredicateFailureReason and errors, you won't need this switch. You can append the failure reason to the second return value and return the error as the third return value:
if failedPredicate, err := c.satisfiesExistingPodsAntiAffinity(pod, meta, nodeInfo); failedPredicate != nil {
failedPredicates := append([]algorithm.PredicateFailureReason{ErrPodAffinityNotMatch), failedPredicate)
return false, failedPredicates, err
}

Reply:
Yes, this is reasonable.